### PR TITLE
TE5-T422 ExpandableListview Done

### PR DESCRIPTION
**Pull request by buddhdev-jay**

### CODE REVIEW

File: app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class GridLayoutRecyclerViewActivity : AppCompatActivity() {
+    var peopleList = ArrayList<People>()
+    var adapter: RecyclerviewGridAdapter? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_grid_layout_recycler_view)
+        addPersonList()
+        adapter = RecyclerviewGridAdapter(peopleList)
+        val layoutManager = GridLayoutManager(this, 2)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt]: static number

File: app/src/main/java/com/example/android_demo_kotlin/MainActivity.kt [hunk 2]
```diff
@@ -34,21 +34,42 @@ class MainActivity : AppCompatActivity() {
...
+        btn_expandable_listview.setOnClickListener {
+            intent = Intent(applicationContext,ExpandableListviewActivity::class.java)
+            startActivity(intent)
+        }
+        btn_bottomnavigation.setOnClickListener {
+            intent = Intent(applicationContext, BottomNavigationActivity::class.java)
+            startActivity(intent)
+        }
+
+        }
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/MainActivity.kt]: code formatting

File: app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableListviewAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class ExpandableListviewAdapter(var context: Context, var headerList: MutableList<String>, var childList: HashMap<String, List<String>>) : BaseExpandableListAdapter() {
+
+    override fun getGroupCount(): Int {
+        return headerList.count()
+    }
+
+    override fun getChildrenCount(position: Int): Int {
+        return this.childList[this.headerList[position]]?.count() ?: SIZE_LIST_ZERO
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableListviewAdapter.kt]: empty line

File: app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+class ExpandableRecyclerAdapter(val AppList: ArrayList<AppDemo>) : RecyclerView.Adapter<ExpandableRecyclerAdapter.ExpandableViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ExpandableViewHolder {
+        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_recyclerview_expandable, parent, false)
+        return ExpandableViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ExpandableViewHolder, position: Int) {
+        holder.apply {
+            nameTxt.text = AppList[position].namehello
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt]: use `with` or `apply` for 
`nameTxt.text = AppList[position].namehello
dateTxt.text = AppList[position].date
apiTxt.text = AppList[position].apilevel
descriptionTxt.text = AppList[position].description`

File: app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    class ExpandableViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        var nameTxt: TextView = itemView.findViewById(R.id.txt_name)
+        var dateTxt: TextView = itemView.findViewById(R.id.txt_date)
+        var apiTxt: TextView = itemView.findViewById(R.id.txt_api)
+        var descriptionTxt: TextView = itemView.findViewById(R.id.description)
+        var linearLayout: LinearLayout = itemView.findViewById(R.id.linearLayout)
+        var expendableLayout: RelativeLayout = itemView.findViewById(R.id.expandable_layout)
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt]: empty line

File: app/src/main/res/layout/item_recyclerview_expandable.xml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/white"
+    app:cardCornerRadius="2dp"
+    app:cardElevation="3dp"
+    app:cardUseCompatPadding="true">
+
+    <LinearLayout
```
Comment (purvesh-d) [inline app/src/main/res/layout/item_recyclerview_expandable.xml]: Why Linear layout used?

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_grid_layout_recycler_view)
+        addPersonList()
+        adapter = RecyclerviewGridAdapter(peopleList)
+        val layoutManager = GridLayoutManager(this, TW0)
+        layoutManager.orientation = LinearLayoutManager.VERTICAL
+        val dividerItemDecoration = DividerItemDecoration(recylerview_grid.context, layoutManager.orientation)
+        recylerview_grid.addItemDecoration(dividerItemDecoration)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt]: use apply

File: app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        holder.apply {
+            AppList[position].apply {
+                nameTxt.text = this.namehello
+                dateTxt.text = this.date
+                apiTxt.text = this.apilevel
+                descriptionTxt.text = this.description
+            }
+            expendableLayout.visibility = if (AppList[position].expandable) View.VISIBLE else View.GONE
+            linearLayout.setOnClickListener {
+                val version = AppList[position]
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/ExpandableRecyclerAdapter.kt]: version var is requierd?

File: app/src/main/res/layout/item_recyclerview_expandable.xml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                android:padding="16dp"
+                android:text="@string/description_text"
+                android:textColor="@color/black"
+                android:textSize="18sp" />
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
```
Comment (purvesh-d) [inline app/src/main/res/layout/item_recyclerview_expandable.xml]: extra empty line

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.android_demo_kotlin.adapter.RecyclerviewGridAdapter
+import com.example.android_demo_kotlin.model.People
+import com.example.android_demo_kotlin.utils.TW0
+import kotlinx.android.synthetic.main.activity_grid_layout_recycler_view.recylerview_grid
+
+class GridLayoutRecyclerViewActivity : AppCompatActivity() {
+    var peopleList = ArrayList<People>()
+    var recycler_adapter: RecyclerviewGridAdapter? = null
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt]: variable name should be in camel case

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt (hunk 10 above)
Comment (buddhdev-jay) [reply app/src/main/java/com/example/android_demo_kotlin/GridLayoutRecyclerViewActivity.kt]: ok i will change

**Pull request merged**